app/models/voucher.rb

```diff
@@ -27,7 +27,12 @@ class Voucher < ApplicationRecord
   validates :voucher_type, inclusion: TYPES
 
   def display_value
-    Spree::Money.new(amount)
+    case voucher_type
+    when FLAT_RATE
+      Spree::Money.new(amount)
+    when PERCENTAGE_RATE
+      I18n.t(:voucher_percentage, amount: amount)
```

Comment:
Why not use Rails helpers? 

* https://apidock.com/rails/v5.2.3/ActionView/Helpers/NumberHelper/number_to_percentage